src/firebolt/utils/util.py

```diff
@@ -181,3 +190,21 @@ def __exit__(
         ):
             log_message = self._message + self.elapsed_time + "s"
             logger.debug(log_message)
+
+
+def parse_url_and_params(url: str) -> Tuple[str, Dict[str, str]]:
+    """Extract URL and query parameters separately from a URL."""
+    url = fix_url_schema(url)
+    parsed_url = urlparse(url)
+    query_params = parse_qs(parsed_url.query)
+    # This strips query parameters from the URL
+    result_url = urljoin(url, parsed_url.path)
+    # parse_qs returns a dictionary with values as lists.
+    # We want the last value in the list.
+    query_params_dict = {}
+    for key, values in query_params.items():
+        # Multiple values for the same key are not expected
+        if len(values) > 1:
+            raise ValueError(f"Multiple values found for key '{key}'")
+        query_params_dict[key] = values[-1]
```

Comment:
nit: values[-1] -> values[0], the same in this context but IMO more straightforward